### PR TITLE
feat(content): Add pagination to TimelineHistory

### PR DESCRIPTION
**Pull request by k88hudson**
This fixes half of #222, the rest is blocked by https://github.com/mozilla/activity-streams/issues/312

### CODE REVIEW

**Comment by pdehaan** [comment]
@k88hudson Needs a rebasin'.

File: lib/ActivityStreams.js [hunk 1]
```diff
@@ -96,25 +96,26 @@ ActivityStreams.prototype = {
    */
   _respondToPlacesRequests(msgName, params) {
     let {msg, worker} = params;
+    const append = msg.meta && msg.meta.append;
```
Comment (oyiptong) [inline lib/ActivityStreams.js]: The memoization key generator will not handle this correctly.

I would suggest to flatten the `msg` object by adding `append` as a parameter of `msg` and removing `meta` altogether, or to improve the cache key generation.

**Comment by oyiptong** [comment]
The code looks OK to me.

The addon bits get an r+ if we take the simple route (using `msg.append` instead of `msg.meta.append`). If we want to improve the cache key generator, we'll need to do more work, and 
I'll take another review when it's ready.

I took a look at the content code and while it looks good to me, I have to admit I'm not familiar with React's conventions and how it will fit in a bigger app. I'll defer to @rlr to review the content side of things.

File: lib/ActivityStreams.js (hunk 1 above)
Comment (oyiptong) [inline lib/ActivityStreams.js]: Actually, I remove my comment. `meta` is not sent as a parameter, so query caching will work correctly

**Comment by oyiptong** [comment]
I took a look at the code, and it looks like `msg.meta` is fine. we only cache based on `params`. The addon code looks good.

**Comment by k88hudson** [comment]
if @rlr isn't available, maybe @pdehaan you could take a look?

**Comment by oyiptong** [comment]
r+

**Pull request merged**